src/Sodium.Core/SodiumLibrary.cs

```diff
@@ -92,7 +92,7 @@ public static partial class SodiumLibrary
     internal static extern int crypto_pwhash_scryptsalsa208sha256_str_verify(byte[] buffer, byte[] password, long passLength);
 
     [DllImport(DllName, CallingConvention = CallingConvention.Cdecl)]
-    internal static extern int crypto_pwhash_str_needs_rehash(byte[] buffer, ulong opsLimit, uint memLimit);
+    internal static extern int crypto_pwhash_str_needs_rehash(byte[] buffer, long opsLimit, int memLimit);
```

Comment:
What's the reasoning for changing the signature of this method? According to the documentation, `opsLimit` is `unsigned long long`, which is `ulong` in C#, while `memLimit` is `size_t`, which is `uint`. 🤨

Reply:
Just for consistency, every method in code base that should be u(long|int) is long and int so i made it consistent with other methods

Reply:
Oh, that's how well I know this codebase.. as I'm not the author, I don't know the reasoning for using signed integers instead of unsigned, that seems odd. Well, your change seems appropriate, then 👍

Reply:
When i refactored codebase a little bit, i ran into problem, i had a lot of casts, so i had two options, change one method to bi consistent or all of them, i wanted to change all of them but it might break the backwards compatibility with existing projects

i think the reason for using longs and ints is that no one will have so large numbers, number of bits in signed and unsigned versions is the same so there will be no loss of data